### DISCUSSION TITLE
Virtualbox split out extension pack into seperate package

### DISCUSSION THREAD

**Discussion by TheCakeIsNaOH** [chocolatey-community/chocolatey-packages#1961] [Ideas]
The `virtualbox` package currently has an option (via a package parameter) for installing the extension pack. This is not following the best practices for packages, since package on the Community Repository are supposed to install only one piece of software at a time. 

Currently, it looks like the checksum for the extension pack is incorrect, which is causing issues, because the only option is to use `--ignore-checksums`, as the `--checksum` would get applied to both virtualbox itself and the extension pack. It also caused issues for a user on discord that had to deal with virtualbox not liking being re-installed on top of a preexisting installation of the same version, while trying to get the extension pack installed.

Therefore, I propose that the extension pack should be split off into it's own package.

**Comment by pauby** [comment]
I'd agree with this for the Community Repository.

**Comment by AdmiringWorm** [comment]
I completely agree. The installation of virtualbox and the extension pack should be in different packages.

I don't remember the exact circumstances why it was added to the main package (simple to do, perhaps), and it may have made sense before Oracle changed their license to require payment if used commercially (which is why it is not being installed by default, initially it had).

As soon as someone is willing to make the needed changes of splitting up the package, we can create an issue based on this discussion they can work against.

**Reply by BobSammers** [reply]
> I don't remember the exact circumstances why it was added to the main package

I have had a brief discussion about this with @pauby on #2137. I remember being quite relieved when these packages were merged in 2016.

> it may have made sense before Oracle changed their license to require payment if used commercially

The extension pack has always been commercial but I assume you are referring to the fact that commercial use was originally allowed free of charge under certain circumstances ("personal use" included commercial users if they downloaded it themselves for their own machine). This freedom wasn't removed until the [release of v5.1.30](https://forums.virtualbox.org/viewtopic.php?t=85092), in Oct 2017, a year _after_ the packages were merged.

My understanding is that the merge was done because the separate packages could not properly be kept in sync and users of both were periodically having problems with installation and upgrade as a result. There are a few comments in the Disqus discussion on the [page for the old, separate extensions package](https://community.chocolatey.org/packages/VirtualBox.ExtensionPack#discussion) about this.

I understand the reasons for seeing these as different packages (and it was the original situation), but my recollection is that errors were much more common before the merge than now so I hope any reversion of this change is done with eyes open to the potential drawbacks.

**Comment by pauby** [comment]
Since we're discussing this over on #2137, I'll respond over there.
